lib/src/source_visitor.dart

```diff
@@ -2723,6 +2723,15 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   @override
   void visitSwitchExpression(SwitchExpression node) {
+    if (node.cases.isEmpty) {
+      // Don't allow splitting an empty switch expression.
+      _visitSwitchValue(node.switchKeyword, node.leftParenthesis,
+          node.expression, node.rightParenthesis);
+      token(node.leftBracket);
+      token(node.rightBracket);
```

Comment:
You'll also want to test how this behaves when the switch is empty but contains comments.

To test/comments/switch.stmt, I would add:

```
>>> line comment
switch (e) {
  // comment
}
<<<
switch (e) {
  // comment
}
>>> line comment on opening line
switch (e) { // comment
}
<<<
switch (e) {
  // comment
}
>>> indented block comment
switch (e) {
  /* comment */
}
<<<
switch (e) {
  /* comment */
}
>>> block comment with trailing newline
switch (e) {/* comment */
}
<<<
switch (e) {
  /* comment */
}
>>> block comment with leading newline
switch (e) {
  /* comment */}
<<<
switch (e) {
  /* comment */
}
>>> inline block comment
switch (e) {  /* comment */  }
<<<
switch (e) {/* comment */}
>>> line comment
e = switch (n) {
  // comment
};
<<<
e = switch (n) {
  // comment
};
>>> line comment on opening line
e = switch (n) { // comment
};
<<<
e = switch (n) {
  // comment
};
>>> indented block comment
e = switch (n) {
  /* comment */
};
<<<
e = switch (n) {
  /* comment */
};
>>> block comment with trailing newline
e = switch (n) {/* comment */
};
<<<
e = switch (n) {
  /* comment */
};
>>> block comment with leading newline
e = switch (n) {
  /* comment */}
<<<
e = switch (n) {
  /* comment */
};
>>> inline block comment
e = switch (n) {  /* comment */  }
<<<
e = switch (n) {/* comment */};
```

You may end up needing to tweak the implementation to get it formatting all of these correctly. (Comments are one of the things that makes formatting hard.)

For comparison, `_visitBody()` (which is used for things like class bodies), has:

```dart
    // Don't allow splitting in an empty body.
    if (nodes.isEmptyBody(rightBracket)) {
      token(leftBracket);
      token(rightBracket);
      return;
    }
```

And `isEmptyBody()` is:

```dart
  bool isEmptyBody(Token rightBracket) =>
      isEmpty && rightBracket.precedingComments == null;
```

That deliberately treats a body containing nothing but a comment as still "non-empty". You may want to do something similar for switches, but then you'll have to make sure to still gracefully handle there being no cases.

Reply:
Good point, thanks!  I'll revise this PR.

Reply:
Ok, added the tests, and fixed the implementation so that they pass.  PTAL.